crates/cairo-coverage/tests/helpers/test_project.rs

```diff
@@ -0,0 +1,105 @@
+use anyhow::{Context, Result};
+use assert_fs::fixture::PathCopy;
+use assert_fs::TempDir;
+use camino::Utf8PathBuf;
+use snapbox::cmd::{cargo_bin, Command as SnapboxCommand};
+use std::fs;
+use std::path::PathBuf;
+
+pub struct TestProject {
+    dir: TempDir,
+}
+
+impl TestProject {
+    pub fn new(test_project_name: &str) -> Result<Self> {
+        let dir = TempDir::new().context("Failed to create a temporary directory")?;
+        dir.copy_from(
+            format!("tests/data/{test_project_name}/"),
+            &["*.toml", "*.cairo"],
+        )
+        .context("Failed to copy project files to the temporary directory")?;
+        Ok(Self { dir })
+    }
+
+    fn generate_trace_files(self) -> Self {
+        SnapboxCommand::new("snforge")
+            .arg("test")
+            .arg("--save-trace-data")
+            .current_dir(&self.dir)
+            .assert()
+            .success();
+        self
+    }
+
+    fn find_trace_files(&self) -> Result<Vec<String>> {
+        let trace_path = self.dir.path().join("snfoundry_trace");
+        fs::read_dir(&trace_path)
+            .context("Failed to read the directory for trace files")?
+            .map(|entry| {
+                entry
+                    .context("Failed to read a directory entry")
+                    .map(|e| e.path().display().to_string())
+            })
+            .collect()
+    }
+
+    fn output_lcov_path(&self) -> PathBuf {
+        self.dir.path().join("coverage.lcov")
+    }
+
+    fn run_coverage(self, args: &[&str]) -> Result<Self> {
+        let trace_files = self.find_trace_files()?;
+        SnapboxCommand::new(cargo_bin!("cairo-coverage"))
+            .args(&trace_files)
+            .args(args)
+            .current_dir(&self.dir)
+            .assert()
+            .success();
+        Ok(self)
+    }
+
+    fn run_genhtml(self) -> Self {
```

Comment:
Is it necessary?

Reply:
This helps because what we output may not have any semantic sense, and `genhtml` verifies that. However, `genhtml` completely depends on the contents of the lcov file. So if the output of the tests doesn't change, the `genhtml` output also won't change. This means that you only need to test the output once when you change it. But then the developer needs to remember to test it with `genhtml` after any changes. Rather than relying on the developer, I would prefer to run this every time to enforce it.

Reply:
It would be nice to add some kind of development guideline to have all the required tools for development listed in one place

Reply:
Will add contribution guide